File: src/explore-education-statistics-common/src/services/types/pageFeedback.ts
export interface PageFeedbackRequest {
  url: string;
  userAgent: string;
  response: PageFeedbackResponse;
  context?: string;
  issue?: string;
  intent?: string;
}

export interface PageFeedbackViewModel {
  id: string;
  created: Date;
  url: string;
  userAgent: string;
  response: PageFeedbackResponse;
  context: string;
  intent: string;
  issue: string;
  read: boolean;
}

export type PageFeedbackResponse =
  | 'Useful'
  | 'NotUseful'
  | 'ProblemEncountered';

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model/PageFeedback.cs
#nullable enable
using GovUk.Education.ExploreEducationStatistics.Common;
using GovUk.Education.ExploreEducationStatistics.Common.Model;

namespace GovUk.Education.ExploreEducationStatistics.Content.Model;

public record PageFeedback : ICreatedTimestamp<DateTime>
{
    public Guid Id { get; set; }

    public DateTime Created { get; set; }

    public required string Url { get; set; }

    public string? UserAgent { get; set; }

    public PageFeedbackResponse Response { get; set; }

    public string? Context { get; set; }

    public string? Issue { get; set; }

    public string? Intent { get; set; }

    public bool Read { get; set; }
}
